plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtMigrateCommandWrapper.java

```diff
@@ -288,6 +295,17 @@ Use VIR_DOMAIN_XML_SECURE (value = 1) prior to v1.0.0.
         return new MigrateAnswer(command, result == null, result, null);
     }
 
+    private int parseAgentPropertiesVmMigrateDomainRetrieveTimeout() throws IOException {
+        final File agentPropertiesFile = PropertiesUtil.findConfigFile(KeyStoreUtils.AGENT_PROPSFILE);
```

Comment:
@GutoVeronezi 
you can use `PropertiesStorage` instead.

Reply:
@weizhouapache I took a look into this class and I am a bit confused. It seems like a facade of `PropertiesUtils` mixed with an another purpouse. 

It seems to me that this class is badly structured, without a clear flow and with nonsense logs. I think it will add a meaningless complexity to the code instead of facility it.

In other hand, in PR https://github.com/apache/cloudstack/pull/4586 there is a proposal of a handler that will simplify this code. If it be approved, I will change this PR to use the handler.